### PR TITLE
Incremental solving producing cprover status as it goes

### PR DESCRIPTION
**Pull request by thk123**
Needs some tidy up but this prints output that is consistent with the
old version

- [x] Each commit message has a non-empty body, explaining why the change was made.
- [x] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [x] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [x] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [na] My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- [x] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

File: src/goto-checker/single_loop_incremental_symex_checker.cpp [hunk 1]
```diff
@@ -47,6 +47,23 @@ single_loop_incremental_symex_checkert::single_loop_incremental_symex_checkert(
...
+  const propertiest &properties,
+  messaget &message_hander)
+{
+  const auto any_failures = std::any_of(
+    properties.begin(),
+    properties.end(),
+    [](const std::pair<irep_idt, property_infot> &property) {
+      return property.second.status == property_statust::FAIL;
+    });
+  std::string status = any_failures ? "FAILURE" : "INCONCLUSIVE";
```
Comment (peterschrammel) [inline src/goto-checker/single_loop_incremental_symex_checker.cpp:60]: How does this behave in the success case?
Please check the output also on a `VERIFICATION SUCCESSFUL` test.

Comment (NlightNFotis) [reply src/goto-checker/single_loop_incremental_symex_checker.cpp:60]: @peterschrammel In the commit message this says that it's not possible to conclude that verification is successful before the unwinding is finished, so it doesn't matter?

File: regression/cbmc-incr-oneloop/multiple-asserts/test.desc [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+Current unwinding: 6\nUnwinding .*\nIncremental status: INCONCLUSIVE
+Current unwinding: 7\nUnwinding .*\nIncremental status: FAILURE
+Current unwinding: 9\nUnwinding .*\nIncremental status: FAILURE
+^EXIT=10$
+^SIGNAL=0$
+^VERIFICATION FAILED$
+--
+^warning: ignoring
+--
+Ensure that with multiple properites, the incremental stauts
```
Comment (NlightNFotis) [inline regression/cbmc-incr-oneloop/multiple-asserts/test.desc]: Can this be changed? 

* `properites` -> `properties`
* `stauts` -> `status`

File: regression/cbmc-incr-oneloop/multiple-asserts/test-no-cp.desc [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+^SIGNAL=0$
+^VERIFICATION FAILED$
+--
+^warning: ignoring
+--
+Ensure that with multiple properites, the incremental stauts
+switches to failure as soon as one property has been falsified
+but keeps unwinding till it has resolved them all
+
+Use `--no-propagation` to ensure that even the output is the same regardless of whether the asserts are checked via the solver or the constant propogator
```
Comment (NlightNFotis) [inline regression/cbmc-incr-oneloop/multiple-asserts/test-no-cp.desc]: `propogator` -> `propagator`

File: regression/cbmc-incr-oneloop/multiple-asserts/test-no-cp.desc [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+Current unwinding: 6\nUnwinding .*\nIncremental status: INCONCLUSIVE
+Current unwinding: 7\nUnwinding .*\nIncremental status: FAILURE
+Current unwinding: 9\nUnwinding .*\nIncremental status: FAILURE
+^EXIT=10$
+^SIGNAL=0$
+^VERIFICATION FAILED$
+--
+^warning: ignoring
+--
+Ensure that with multiple properites, the incremental stauts
```
Comment (NlightNFotis) [inline regression/cbmc-incr-oneloop/multiple-asserts/test-no-cp.desc]: ditto

**Review by NlightNFotis — APPROVED** [review]
Minor faults, it's good to go once they are fixed as far as I am concerned.

**Comment by thk123** [comment]
@peterschrammel @NlightNFotis comments addressed -> it required a little tweaking to exactly where the printing went to get it working correctly but I'm happy with it now. 

The behaviour is:

In each step, before doing the next round of unwinding, but after solving the equation, we will print inconclusive if there no failed properties yet, otherwise we'll print failed. Once we've either got an answer to all properties, or finished unwinding, no incremental status will be reported, since the final status will be reported next.

**Review by peterschrammel — APPROVED** [review]

**Pull request merged**